// @telegram-apps/telegram-ui/dist/components/Form/PinInput/hooks/usePinInput.js
'use client';
import { useCallback, useRef } from "react";
import { Keys } from "../../../../helpers/accessibility";
import { clamp } from "../../../../helpers/math";
import { useCustomEnsuredControl } from "../../../../hooks/useEnsureControl";
export var BiometricType;
(function(BiometricType) {
    BiometricType["FACEID"] = "faceid";
    BiometricType["FINGERPRINT"] = "fingerprint";
})(BiometricType || (BiometricType = {}));
const BIOMETRIC_AUTH_BUTTON_VALUE = 'BiometricAuth';
export const DEFAULT_PINS = [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    BIOMETRIC_AUTH_BUTTON_VALUE,
    0,
    Keys.BACKSPACE
];
export const getAvailablePins = (biometricType)=>{
    const pins = [
        ...DEFAULT_PINS
    ];
    if (!biometricType) {
        const biometricIndex = pins.indexOf(BIOMETRIC_AUTH_BUTTON_VALUE);
        if (biometricIndex !== -1) {
            pins.splice(biometricIndex, 1);
        }
    }
    return pins;
};
export const usePinInput = ({ pinCount, value: valueProp = [], onChange, onBiometricAuth, biometricType = undefined })=>{
    const PINS = getAvailablePins(biometricType);
    const inputRefs = useRef([]).current;
    const [value, setValue] = useCustomEnsuredControl({
        defaultValue: valueProp,
        onChange
    });
    const focusByIndex = (index)=>{
        requestAnimationFrame(()=>{
            var _inputRefs_index;
            return (_inputRefs_index = inputRefs[index]) === null || _inputRefs_index === void 0 ? void 0 : _inputRefs_index.focus();
        });
    };
    const setValueByIndex = (index, newValue)=>{
        setValue((prev)=>{
            const nextValue = [
                ...prev
            ];
            nextValue[index] = newValue;
            return nextValue;
        });
    };
    const removeLastValue = (currentIndex)=>{
        setValue((prev)=>prev.slice(0, -1));
        focusByIndex(currentIndex - 1);
    };
    const handleClickValue = useCallback((enteredValue)=>{
        const lastIndex = clamp(value.length, 0, pinCount - 1);
        setValueByIndex(lastIndex, enteredValue);
        focusByIndex(lastIndex + 1);
    }, [
        value,
        pinCount
    ]);
    const handleClickBackspace = useCallback(()=>{
        removeLastValue(value.length - 1);
    }, [
        value
    ]);
    const handleBiometricAuth = useCallback(()=>{
        if (onBiometricAuth) {
            onBiometricAuth();
        }
    }, [
        onBiometricAuth
    ]);
    const handleButton = useCallback((index, button)=>{
        const numButton = Number(button);
        if (!Number.isNaN(numButton) && PINS.includes(numButton)) {
            setValueByIndex(index, numButton);
            focusByIndex(index + 1);
        }
        switch(button){
            case Keys.BACKSPACE:
                removeLastValue(index);
                break;
            case Keys.ARROW_LEFT:
                focusByIndex(index - 1);
                break;
            case Keys.ARROW_RIGHT:
                focusByIndex(index + 1);
                break;
            default:
                break;
        }
    }, []);
    const setInputRefByIndex = useCallback((index, ref)=>{
        if (!ref) {
            return;
        }
        inputRefs[index] = ref;
    }, []);
    const onElementClick = useCallback((element)=>{
        if (element === Keys.BACKSPACE) {
            handleClickBackspace();
            return;
        }
        if (element === BIOMETRIC_AUTH_BUTTON_VALUE) {
            handleBiometricAuth();
            return;
        }
        handleClickValue(Number(element));
    }, [
        handleClickValue,
        handleClickBackspace,
        handleBiometricAuth
    ]);
    return {
        value,
        setInputRefByIndex,
        onElementClick,
        handleButton,
        PINS
    };
};
