// @telegram-apps/telegram-ui/dist/cjs/components/Navigation/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Breadcrumbs: function() {
        return _Breadcrumbs.Breadcrumbs;
    },
    CompactPagination: function() {
        return _CompactPagination.CompactPagination;
    },
    Link: function() {
        return _Link.Link;
    },
    Pagination: function() {
        return _Pagination.Pagination;
    },
    SegmentedControl: function() {
        return _SegmentedControl.SegmentedControl;
    },
    TabsList: function() {
        return _TabsList.TabsList;
    }
});
const _Breadcrumbs = require("./Breadcrumbs/Breadcrumbs");
const _CompactPagination = require("./CompactPagination/CompactPagination");
const _Link = require("./Link/Link");
const _Pagination = require("./Pagination/Pagination");
const _SegmentedControl = require("./SegmentedControl/SegmentedControl");
const _TabsList = require("./TabsList/TabsList");
