// @telegram-apps/telegram-ui/dist/cjs/components/Form/PinInput/hooks/usePinInput.js
'use client';
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    BiometricType: function() {
        return BiometricType;
    },
    DEFAULT_PINS: function() {
        return DEFAULT_PINS;
    },
    getAvailablePins: function() {
        return getAvailablePins;
    },
    usePinInput: function() {
        return usePinInput;
    }
});
const _react = require("react");
const _accessibility = require("../../../../helpers/accessibility");
const _math = require("../../../../helpers/math");
const _useEnsureControl = require("../../../../hooks/useEnsureControl");
var BiometricType;
(function(BiometricType) {
    BiometricType["FACEID"] = "faceid";
    BiometricType["FINGERPRINT"] = "fingerprint";
})(BiometricType || (BiometricType = {}));
const BIOMETRIC_AUTH_BUTTON_VALUE = 'BiometricAuth';
const DEFAULT_PINS = [
    1,
    2,
    3,
    4,
    5,
    6,
    7,
    8,
    9,
    BIOMETRIC_AUTH_BUTTON_VALUE,
    0,
    _accessibility.Keys.BACKSPACE
];
const getAvailablePins = (biometricType)=>{
    const pins = [
        ...DEFAULT_PINS
    ];
    if (!biometricType) {
        const biometricIndex = pins.indexOf(BIOMETRIC_AUTH_BUTTON_VALUE);
        if (biometricIndex !== -1) {
            pins.splice(biometricIndex, 1);
        }
    }
    return pins;
};
const usePinInput = ({ pinCount, value: valueProp = [], onChange, onBiometricAuth, biometricType = undefined })=>{
    const PINS = getAvailablePins(biometricType);
    const inputRefs = (0, _react.useRef)([]).current;
    const [value, setValue] = (0, _useEnsureControl.useCustomEnsuredControl)({
        defaultValue: valueProp,
        onChange
    });
    const focusByIndex = (index)=>{
        requestAnimationFrame(()=>{
            var _inputRefs_index;
            return (_inputRefs_index = inputRefs[index]) === null || _inputRefs_index === void 0 ? void 0 : _inputRefs_index.focus();
        });
    };
    const setValueByIndex = (index, newValue)=>{
        setValue((prev)=>{
            const nextValue = [
                ...prev
            ];
            nextValue[index] = newValue;
            return nextValue;
        });
    };
    const removeLastValue = (currentIndex)=>{
        setValue((prev)=>prev.slice(0, -1));
        focusByIndex(currentIndex - 1);
    };
    const handleClickValue = (0, _react.useCallback)((enteredValue)=>{
        const lastIndex = (0, _math.clamp)(value.length, 0, pinCount - 1);
        setValueByIndex(lastIndex, enteredValue);
        focusByIndex(lastIndex + 1);
    }, [
        value,
        pinCount
    ]);
    const handleClickBackspace = (0, _react.useCallback)(()=>{
        removeLastValue(value.length - 1);
    }, [
        value
    ]);
    const handleBiometricAuth = (0, _react.useCallback)(()=>{
        if (onBiometricAuth) {
            onBiometricAuth();
        }
    }, [
        onBiometricAuth
    ]);
    const handleButton = (0, _react.useCallback)((index, button)=>{
        const numButton = Number(button);
        if (!Number.isNaN(numButton) && PINS.includes(numButton)) {
            setValueByIndex(index, numButton);
            focusByIndex(index + 1);
        }
        switch(button){
            case _accessibility.Keys.BACKSPACE:
                removeLastValue(index);
                break;
            case _accessibility.Keys.ARROW_LEFT:
                focusByIndex(index - 1);
                break;
            case _accessibility.Keys.ARROW_RIGHT:
                focusByIndex(index + 1);
                break;
            default:
                break;
        }
    }, []);
    const setInputRefByIndex = (0, _react.useCallback)((index, ref)=>{
        if (!ref) {
            return;
        }
        inputRefs[index] = ref;
    }, []);
    const onElementClick = (0, _react.useCallback)((element)=>{
        if (element === _accessibility.Keys.BACKSPACE) {
            handleClickBackspace();
            return;
        }
        if (element === BIOMETRIC_AUTH_BUTTON_VALUE) {
            handleBiometricAuth();
            return;
        }
        handleClickValue(Number(element));
    }, [
        handleClickValue,
        handleClickBackspace,
        handleBiometricAuth
    ]);
    return {
        value,
        setInputRefByIndex,
        onElementClick,
        handleButton,
        PINS
    };
};
