// @telegram-apps/telegram-ui/dist/cjs/components/Blocks/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Accordion: function() {
        return _Accordion.Accordion;
    },
    Avatar: function() {
        return _Avatar.Avatar;
    },
    AvatarStack: function() {
        return _AvatarStack.AvatarStack;
    },
    Badge: function() {
        return _Badge.Badge;
    },
    Banner: function() {
        return _Banner.Banner;
    },
    Blockquote: function() {
        return _Blockquote.Blockquote;
    },
    Button: function() {
        return _Button.Button;
    },
    ButtonCell: function() {
        return _ButtonCell.ButtonCell;
    },
    Card: function() {
        return _Card.Card;
    },
    Cell: function() {
        return _Cell.Cell;
    },
    IconButton: function() {
        return _IconButton.IconButton;
    },
    IconContainer: function() {
        return _IconContainer.IconContainer;
    },
    Image: function() {
        return _Image.Image;
    },
    Info: function() {
        return _Info.Info;
    },
    InlineButtons: function() {
        return _InlineButtons.InlineButtons;
    },
    List: function() {
        return _List.List;
    },
    Navigation: function() {
        return _Navigation.Navigation;
    },
    Placeholder: function() {
        return _Placeholder.Placeholder;
    },
    Section: function() {
        return _Section.Section;
    },
    Steps: function() {
        return _Steps.Steps;
    },
    Timeline: function() {
        return _Timeline.Timeline;
    }
});
const _Accordion = require("./Accordion/Accordion");
const _Avatar = require("./Avatar/Avatar");
const _AvatarStack = require("./AvatarStack/AvatarStack");
const _Badge = require("./Badge/Badge");
const _Banner = require("./Banner/Banner");
const _Blockquote = require("./Blockquote/Blockquote");
const _Button = require("./Button/Button");
const _Card = require("./Card/Card");
const _Cell = require("./Cell/Cell");
const _ButtonCell = require("./Cell/components/ButtonCell/ButtonCell");
const _Info = require("./Cell/components/Info/Info");
const _Navigation = require("./Cell/components/Navigation/Navigation");
const _IconButton = require("./IconButton/IconButton");
const _IconContainer = require("./IconContainer/IconContainer");
const _Image = require("./Image/Image");
const _InlineButtons = require("./InlineButtons/InlineButtons");
const _List = require("./List/List");
const _Placeholder = require("./Placeholder/Placeholder");
const _Section = require("./Section/Section");
const _Steps = require("./Steps/Steps");
const _Timeline = require("./Timeline/Timeline");
