// @telegram-apps/telegram-ui/dist/cjs/components/Feedback/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    CircularProgress: function() {
        return _CircularProgress.CircularProgress;
    },
    Progress: function() {
        return _Progress.Progress;
    },
    Skeleton: function() {
        return _Skeleton.Skeleton;
    },
    Snackbar: function() {
        return _Snackbar.Snackbar;
    },
    Spinner: function() {
        return _Spinner.Spinner;
    },
    Spoiler: function() {
        return _Spoiler.Spoiler;
    }
});
const _CircularProgress = require("./CircularProgress/CircularProgress");
const _Progress = require("./Progress/Progress");
const _Skeleton = require("./Skeleton/Skeleton");
const _Snackbar = require("./Snackbar/Snackbar");
const _Spinner = require("./Spinner/Spinner");
const _Spoiler = require("./Spoiler/Spoiler");
