// @telegram-apps/telegram-ui/dist/cjs/components/Form/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Checkbox: function() {
        return _Checkbox.Checkbox;
    },
    Chip: function() {
        return _Chip.Chip;
    },
    ColorInput: function() {
        return _ColorInput.ColorInput;
    },
    FileInput: function() {
        return _FileInput.FileInput;
    },
    Input: function() {
        return _Input.Input;
    },
    Multiselect: function() {
        return _Multiselect.Multiselect;
    },
    Multiselectable: function() {
        return _Multiselectable.Multiselectable;
    },
    PinInput: function() {
        return _PinInput.PinInput;
    },
    Radio: function() {
        return _Radio.Radio;
    },
    Rating: function() {
        return _Rating.Rating;
    },
    Select: function() {
        return _Select.Select;
    },
    Selectable: function() {
        return _Selectable.Selectable;
    },
    Slider: function() {
        return _Slider.Slider;
    },
    Switch: function() {
        return _Switch.Switch;
    },
    Textarea: function() {
        return _Textarea.Textarea;
    }
});
const _Checkbox = require("./Checkbox/Checkbox");
const _Chip = require("./Chip/Chip");
const _ColorInput = require("./ColorInput/ColorInput");
const _FileInput = require("./FileInput/FileInput");
const _Input = require("./Input/Input");
const _Multiselect = require("./Multiselect/Multiselect");
const _Multiselectable = require("./Multiselectable/Multiselectable");
const _PinInput = require("./PinInput/PinInput");
const _Radio = require("./Radio/Radio");
const _Rating = require("./Rating/Rating");
const _Select = require("./Select/Select");
const _Selectable = require("./Selectable/Selectable");
const _Slider = require("./Slider/Slider");
const _Switch = require("./Switch/Switch");
const _Textarea = require("./Textarea/Textarea");
