// @telegram-apps/telegram-ui/dist/cjs/components/Typography/index.js
"use strict";
Object.defineProperty(exports, "__esModule", {
    value: true
});
function _export(target, all) {
    for(var name in all)Object.defineProperty(target, name, {
        enumerable: true,
        get: all[name]
    });
}
_export(exports, {
    Caption: function() {
        return _Caption.Caption;
    },
    Headline: function() {
        return _Headline.Headline;
    },
    LargeTitle: function() {
        return _LargeTitle.LargeTitle;
    },
    Subheadline: function() {
        return _Subheadline.Subheadline;
    },
    Text: function() {
        return _Text.Text;
    },
    Title: function() {
        return _Title.Title;
    },
    Typography: function() {
        return _Typography.Typography;
    }
});
const _Caption = require("./Caption/Caption");
const _Headline = require("./Headline/Headline");
const _LargeTitle = require("./LargeTitle/LargeTitle");
const _Subheadline = require("./Subheadline/Subheadline");
const _Text = require("./Text/Text");
const _Title = require("./Title/Title");
const _Typography = require("./Typography");
